### PR TITLE
Remove easy uses of setAccessible in tests.

### PR DESCRIPTION
**Pull request by rmuir**
Don't worry, I will fix the rest. But some of those remaining will need a lucene upgrade,
we need to add a getter or two for tests to do things cleanly.

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -293,6 +293,8 @@
                                 <include>org/elasticsearch/common/util/MockBigArrays$*.class</include>
                                 <include>org/elasticsearch/node/NodeMocksPlugin.class</include>
                                 <include>org/elasticsearch/node/MockNode.class</include>
+                                <include>org/elasticsearch/node/MockNode.class</include>
```
Comment (rjernst) [inline core/pom.xml]: My is mock node duplicated?

Comment (rmuir) [inline core/pom.xml]: I committed a fix for this

File: core/src/test/java/org/elasticsearch/indices/analysis/PreBuiltAnalyzerIntegrationIT.java [hunk 2]
```diff
@@ -161,44 +162,18 @@ private void assertThatAnalyzersHaveBeenLoaded(Map<PreBuiltAnalyzers, List<Versi
         }
     }
 
-    // the close() method of a lucene analyzer sets the storedValue field to null
-    // we simply check this via reflection - ugly but works
-    private void assertLuceneAnalyzersAreNotClosed(Map<PreBuiltAnalyzers, List<Version>> loadedAnalyzers) throws IllegalAccessException, NoSuchFieldException {
+    // ensure analyzers are still open by checking there is no ACE
```
Comment (rjernst) [inline core/src/test/java/org/elasticsearch/indices/analysis/PreBuiltAnalyzerIntegrationIT.java]: nice simplification

**Comment by rjernst** [comment]
LGTM

**Comment by rmuir** [comment]
I just backported this to 2.x

**Pull request merged**